thavalon-webapp/src/utils/GameSocket.tsx

```diff
@@ -0,0 +1,67 @@
+enum WEBSOCKET_READYSTATE {
+    CONNECTING = 0,
+    OPEN = 1,
+    CLOSING = 2,
+    CLOSED = 3,
+};
+
+
+interface PingResponse {
+    "result": boolean,
+    "errorMessage": string, // only populated if result is false
+}
+
+export class GameSocket {
+    // the instantiated instance of the gamesocket
+    private static instance: GameSocket;
+    // the instantiated instance of the underlying websocket
+    private static websocket: WebSocket;

```

Comment:
Does this need to be `static`?

Reply:
hmm probably shouldn't, in case we ever need to make a new websocket due to connection issues or something